commcare_connect/data_export/views.py

```
@@ -401,6 +401,19 @@ def _check_edit_permissions(self, request):
                 programs.add(item["program_id"])
             if item.get("organization_id"):
                 orgs.add(item["organization_id"])
+            if not any(item.get(k) for k in ("opportunity_id", "program_id", "organization_id")) and item.get("id"):
+                # No explicit scope provided — resolve ownership from the existing record so that a
+                # bare {"id": N} cannot bypass the ownership check and target any record by raw PK.
+                try:
+                    record = LabsRecord.objects.get(pk=item["id"])
+                except LabsRecord.DoesNotExist:
+                    raise NotFound()
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)